web_server.go

```diff
@@ -58,15 +61,21 @@ func newWebServer(
 		return nil, fmt.Errorf("failed reading default index.html page: %v", err)
 	}
 
-	fs := handleStatic([]byte(mungeIndexPage(baseURL, string(indexPage))), http.StripPrefix(baseURL, http.FileServer(http.Dir(webDir))), header)
+	fs := handleStatic([]byte(mungeIndexPage(baseURL, string(indexPage), previewImage, previewTitle, previewDescription)), http.StripPrefix(baseURL, http.FileServer(http.Dir(webDir))), header)
 	if auth != nil {
 		fs = auth.wrap(fs)
 	}
 	return &WebServer{site, broker, fs, keychain, maxRequestSize, baseURL}, nil
 }
 
-func mungeIndexPage(baseURL, html string) string {
+func mungeIndexPage(baseURL, html string, previewImage string, previewTitle string, previewDescription string) string {
 	// HACK
+	// add meta tags for preview image, title, description to the html string
+	metaTags := fmt.Sprintf(`<meta property="og:image" content="%s">
+		<meta property="og:title" content="%s">
+		<meta property="og:description" content="%s">`, previewImage, previewTitle, previewDescription)
```

Comment:
Are these enough to work across all the mainstream networks like Twitter, Linkedin, Facebook, Medium etc.?

Reply:
Yes! I just learned about Open Graph Protocol for this and it's supported by most modern platforms.

Reply:
Perfect!